packages/flutter_tools/lib/src/version.dart

```diff
@@ -24,7 +24,7 @@ enum Channel {
 }
 
 /// The flutter GitHub repository.
-const String _flutterGit = 'https://github.com/flutter/flutter.git';
+final String _flutterGit = globals.platform.environment['FLUTTER_GIT_URL'] ?? 'https://github.com/flutter/flutter.git';
```

Comment:
Unfortunately this isn't really testable since it can't be changed after the first time it's set.  I wrote a unit test for this that passed on its own, but failed when run with the entire test suite because it always returned `https://github.com/flutter/flutter.git` because a previous test had initialized it.

How about:
```dart
String get _flutterGit => globals.platform.environment['FLUTTER_GIT_URL'] ??
    'https://github.com/flutter/flutter.git';
```

Reply:
Good catch